src/main/kotlin/step2/StringExpressionParser.kt

```diff
@@ -0,0 +1,29 @@
+package step2
+
+object StringExpressionParser {
+    private val FORMULA_REGEX = Regex("\\d([+|\\-|*|/]\\d)*")
+    fun parse(input: String): Expression? {
+        if (input.isEmpty() || FORMULA_REGEX.matchEntire(input) == null) {
+            throw IllegalArgumentException("입력된 문자열이 올바르지 않습니다. $input")
+        }
+        return parseFormula(input)
+    }
+
+    private fun parseFormula(formulaStr: String): Expression? {
+        var expression: Expression? = null
+        for (i: Int in formulaStr.length - 1 downTo 1) {
+            if (i % 2 == 1) {
+                continue
+            }
```

Comment:
두 단계씩 i 값을 변경시키는 것이 의도이시라면, 아래 코틀린 기능이 도움이 되실겁니다 :)
```suggestion
        for(i in formulaStr.length - 1 downTo 1 step 2)
```